tests/io/test_kedro_data_catalog.py

```diff
@@ -0,0 +1,658 @@
+
+    class TestKedroDataCatalogFromConfig:
+        def test_from_correct_config(self, data_catalog_from_config, dummy_dataframe):
+            """Test populating the data catalog from config"""
+            data_catalog_from_config.save("boats", dummy_dataframe)
+            reloaded_df = data_catalog_from_config.load("boats")
+            assert_frame_equal(reloaded_df, dummy_dataframe)
+
+        def test_config_missing_type(self, correct_config):
+            """Check the error if type attribute is missing for some data set(s)
+            in the config"""
+            del correct_config["catalog"]["boats"]["type"]
+            pattern = (
+                "An exception occurred when parsing config for dataset 'boats':\n"
+                "'type' is missing from dataset catalog configuration"
+            )
+            with pytest.raises(DatasetError, match=re.escape(pattern)):
+                KedroDataCatalog.from_config(**correct_config)
+
+        def test_config_invalid_module(self, correct_config):
+            """Check the error if the type points to nonexistent module"""
+            correct_config["catalog"]["boats"]["type"] = (
+                "kedro.invalid_module_name.io.CSVDataset"
+            )
+
+            error_msg = "Class 'kedro.invalid_module_name.io.CSVDataset' not found"
+            with pytest.raises(DatasetError, match=re.escape(error_msg)):
+                KedroDataCatalog.from_config(**correct_config)
+
+        def test_config_relative_import(self, correct_config):
+            """Check the error if the type points to a relative import"""
+            correct_config["catalog"]["boats"]["type"] = ".CSVDatasetInvalid"
+
+            pattern = "'type' class path does not support relative paths"
+            with pytest.raises(DatasetError, match=re.escape(pattern)):
+                KedroDataCatalog.from_config(**correct_config)
+
+        def test_config_import_kedro_datasets(self, correct_config, mocker):
+            """Test kedro_datasets default path to the dataset class"""
+            # Spy _load_obj because kedro_datasets is not installed and we can't import it.
+
+            import kedro.io.core
+
+            spy = mocker.spy(kedro.io.core, "_load_obj")
+            parse_dataset_definition(correct_config["catalog"]["boats"])
+            for prefix, call_args in zip(_DEFAULT_PACKAGES, spy.call_args_list):
+                # In Python 3.7 call_args.args is not available thus we access the call
+                # arguments with less meaningful index.
+                # The 1st index returns a tuple, the 2nd index return the name of module.
+                assert call_args[0][0] == f"{prefix}pandas.CSVDataset"
+
+        def test_config_import_extras(self, correct_config):
+            """Test kedro_datasets default path to the dataset class"""
+            correct_config["catalog"]["boats"]["type"] = "pandas.CSVDataset"
+            assert KedroDataCatalog.from_config(**correct_config)
```

Comment:
This looks like an old test when datasets still lived in "extras", can we remove it?